### PR TITLE
always pull guest kernel but don't build

### PR DESCRIPTION
**Pull request by il-steffen**
Always pull linux-guest so we have something to work with.
Only leave configure/build to the user workflow.

### CODE REVIEW

**Comment by Wenzel** [comment]
I would propose to conditionaly include `build.yml`
~~~yaml
- name: Clone linux-guest repo
  git:
    repo: "{{ guest_url }}"
    dest: "{{ guest_root }}"
    version: "{{ guest_revision | default(omit) }}"
    depth: "{{ git_clone_depth | default(omit) }}"
    force: yes

- name: Build the TDX guest kernel
  include_tasks: build.yml
  tags:
    - guest-build
    - never
~~~

File: deploy/roles/guest/tasks/main.yml [hunk 1]
```diff
@@ -17,21 +17,8 @@
...
-    - ./scripts/config -e CONFIG_TDX_FUZZ_KAFL
-    - ./scripts/config -e CONFIG_TDX_FUZZ_HARNESS_DOINITCALLS_VIRTIO
-
-- name: Build TDX guest kernel
-  make:
-    chdir: "{{ guest_root }}"
-    params:
-      --jobs: "{{ ansible_processor_nproc }}"
+- name: Prepare fuzzing build of TDX guest kernel
+  include_tasks: build.yml
```
Comment (Wenzel) [inline deploy/roles/guest/tasks/main.yml]: I just tested your PR, and I realized that the tasks in `build.yml` were not executed.

The solution is to replace `include_tasks` with [`import_tasks`](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/import_tasks_module.html) which is evaluated statically before the playbook is executed.

**Review by Wenzel — CHANGES_REQUESTED** [review]

**Review by Wenzel — APPROVED** [review]

File: deploy/roles/guest/tasks/main.yml (hunk 1 above)
Comment (il-steffen) [reply deploy/roles/guest/tasks/main.yml]: I could not reproduce, this seemed to work fine with either option:
`make deploy --  --tags guest,guest_build`

But indeed, import_tasks seems more appropriate when just splitting up playbooks for convenience.

Comment (Wenzel) [reply deploy/roles/guest/tasks/main.yml]: On my side, this is what happens when I used the `include_tasks` version:
`make deploy -- --tags guest_build`
image

**Pull request merged**